src/backend/distributed/commands/alter_table.c

```diff
@@ -1726,6 +1726,17 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 		ExecuteQueryViaSPI(justBeforeDropCommand, SPI_OK_UTILITY);
 	}
 
+	if (!suppressNoticeMessages)
+	{
+		ereport(NOTICE, (errmsg("analyzing the new %s",
```

Comment:
It seems like breaking lots of tests.

It is mainly due to `Analyze` is not compatible with `local execution`. However, `undistribute_table` or `alter_distributed_table` could use local execution. And, it is not allowed to change state of the local execution within a transaction block.

See test failures: https://app.circleci.com/pipelines/github/citusdata/citus/32533/workflows/15de3182-ff86-48c7-a489-3b1b483019f9/jobs/1116145

Reply:
In the code, we say that local execution is NOT allowed for `VACUUM`, which is still valid.

But, for `Analyze` alone, we could relax the rule to allow local execution: https://github.com/citusdata/citus/blob/main/src/backend/distributed/commands/vacuum.c#L194-L197

Can you add that as a separate commit.

Reply:
Thanks for the pointers! I am going to close this PR because of irrelevancy